src/_pytest/reports.py

```diff
@@ -190,11 +190,17 @@ def head_line(self) -> str | None:
             return domain
         return None
 
-    def _get_verbose_word(self, config: Config):
+    def _get_verbose_word_with_markup(
+        self, config: Config, default_markup: Mapping[str, bool]
+    ) -> tuple[str, Mapping[str, bool]]:
         _category, _short, verbose = config.hook.pytest_report_teststatus(
             report=self, config=config
         )
-        return verbose
+
+        if isinstance(verbose, str):
+            return verbose, default_markup
+
+        return verbose[0], verbose[1]
```

Comment:
While at it, we should probably also verify that verbose what we expect and raise an error otherwise:

```suggestion
        if isinstance(verbose, Sequence) and len(verbose) == 2:
            word, markup = verbose
            if isinstance(word, str) and isinstance(markup, Mapping):
                return word, markup
        pytest.fail("pytest_report_teststatus() hook (from a plugin) returned an invalid verbose value: {verbose!r}.\nExpected either a string or a tuple of (word, markup)."
```

Reply:
(being able to use `match` here would be awesome, but alas only Python 3.10+).